pennylane/workflow/construct_batch.py

```diff
@@ -0,0 +1,268 @@
+# Copyright 2018-2024 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Contains a function extracting the tapes at postprocessing at any stage of a transform program.
+
+"""
+import inspect
+from typing import Union, Callable, Tuple
+
+import pennylane as qml
+from .qnode import QNode, _make_execution_config, _get_device_shots
+
+
+def _get_full_transform_program(qnode: QNode) -> "qml.transforms.core.TransformProgram":
+    program = qml.transforms.core.TransformProgram(qnode.transform_program)
+    if getattr(qnode.gradient_fn, "expand_transform", False):
+        program.add_transform(
+            qml.transform(qnode.gradient_fn.expand_transform),
+            **qnode.gradient_kwargs,
+        )
+    if isinstance(qnode.device, qml.devices.Device):
+        config = _make_execution_config(qnode)
+        return program + qnode.device.preprocess(config)[0]
+    program.add_transform(qml.transform(qnode.device.batch_transform))
+    program.add_transform(qml.transforms.core.expand_fn_transform(qnode.device.expand_fn))
+    return program
+
+
+def transform_program(qnode: "QNode", level=None) -> "qml.transforms.core.TransformProgram":
```

Comment:
I wonder if this could be `get_transform_program` instead? To me `transform_program` sounds like a function that constructs a `TransformProgram`, rather than retrieving one from a `qnode`.

Reply:
That was something I was worrying about too.  I'll switch to `get_transform_program`.